astropy/io/fits/tests/test_connect.py

```diff
@@ -64,6 +65,15 @@ def setup_class(self):
             dtype=[("a", int), ("b", "U1"), ("c", float)],
         )
 
+    def test_path(self, tmp_path):
+        filename = "temp.fits"
+        t1 = Table(self.data)
+        t1.write(filename, overwrite=True)
+        try:
+            t1.write(Path(filename), format="fits", overwrite=True)
+        except OSError as error:
+            pytest.fail(f"overwrite true raises {error}")
```

Comment:
`filename` is already a path using `tmp_path`. No need to explicitly catch the exception because any exception or unhandled warning is going to fail the test anyway.

```suggestion
        t1.write(filename, format="fits", overwrite=True)
```

Reply:
Same comment as above ... for some reason just using `filename` does not make the test fail on main...